e2e/cypress/integration/integrations/incoming_webhook/webhook_posts_when_creator_not_in_channel_spec.js

```diff
@@ -0,0 +1,97 @@
+    let testChannel;
+    let newIncomingHook;
+    let incomingWebhook;
+
+    before(() => {
+        // # Create new setup
+        cy.apiInitSetup().then(({user}) => {
+            testUser = user;
+
+            // # Creaate a second user
+            cy.apiCreateUser().then(({user: user2}) => {
+                secondUser = user2;
+            });
+
+            // # Login as the new user
+            cy.apiLogin(testUser).then(() => {
+                // # Create a new team with the new user
+                cy.apiCreateTeam('test-team', 'Team Testers').then(({team}) => {
+                    testTeam = team;
+
+                    // # Add second user to the test team
+                    cy.apiAddUserToTeam(testTeam.id, secondUser.id);
+
+                    // # Create a new test channel for the team
+                    cy.apiCreateChannel(testTeam.id, 'test-channel', 'Testers Channel').then(({channel}) => {
+                        testChannel = channel;
+
+                        newIncomingHook = {
+                            channel_id: testChannel.id,
+                            channel_locked: true,
+                            description: 'Test Webhook Description',
+                            display_name: 'Test Webhook Name',
+                        };
+
+                        //# Create a new webhook
+                        cy.apiCreateWebhook(newIncomingHook).then((hook) => {
+                            incomingWebhook = hook;
+                        });
+
+                        // # Add second user to the test channel
+                        cy.apiAddUserToChannel(testChannel.id, secondUser.id).then(() => {
+                            // # Remove the first user from the channel
+                            cy.apiDeleteUserFromTeam(testTeam.id, testUser.id).then(({data}) => {
+                                expect(data.status).to.equal('OK');
+                            });
+
+                            // # Login as the second user
+                            cy.apiLogin(secondUser);
+
+                            // # Visit the test channel
+                            cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+                        });
+                    });
+                });
+            });
+        });
+    });
+
+    it('MM-T638 Webhook posts when webhook creator is not a member of the channel', () => {
+        const payload = getPayload(testChannel);
+
+        // # Post the webhook message
+        cy.postIncomingWebhook({url: incomingWebhook.url, data: payload});
+
+        // * Assert that the message was posted even though webhook author has been removed
+        cy.uiWaitUntilMessagePostedIncludes('this webhook was set up by a used that is no longer in this channel').then(() => {
+            cy.getLastPostId().then((postId) => {
+                cy.get(`#postMessageText_${postId}`).should('have.text', `${payload.text}`);
+            });
+        });
```

Comment:
- can use `payload.text` instead of repeating hard-coded string
- `cy.getLastPostId` does not need to put inside the block
```suggestion
        cy.uiWaitUntilMessagePostedIncludes(payload.text);
        cy.getLastPostId().then((postId) => {
            cy.get(`#postMessageText_${postId}`).should('have.text', payload.text);
        });
```